CI fails for experimental tests: RuntimeError: Some tensors share memory

CI fails for experimental tests: https://github.com/huggingface/trl/actions/runs/22298532846/job/64500461447?pr=5148
> RuntimeError: 
>              Some tensors share memory, this will lead to duplicate memory on disk and potential differences when loading them again

```python
  FAILED tests/experimental/test_ppo_trainer.py::TestPPOTrainer::test_peft_training - RuntimeError: 
              Some tensors share memory, this will lead to duplicate memory on disk and potential differences when loading them again: [{'value_model.model.embed_tokens.weight', 'critic_backbone.embed_tokens.weight'}, {'critic_backbone.layers.0.self_attn.q_proj.weight', 'value_model.model.layers.0.self_attn.q_proj.weight'}, {'critic_backbone.layers.0.self_attn.q_proj.bias', 'value_model.model.layers.0.self_attn.q_proj.bias'}, {'value_model.model.layers.0.self_attn.k_proj.weight', 'critic_backbone.layers.0.self_attn.k_proj.weight'}, {'critic_backbone.layers.0.self_attn.k_proj.bias', 'value_model.model.layers.0.self_attn.k_proj.bias'}, {'critic_backbone.layers.0.self_attn.v_proj.weight', 'value_model.model.layers.0.self_attn.v_proj.weight'}, {'critic_backbone.layers.0.self_attn.v_proj.bias', 'value_model.model.layers.0.self_attn.v_proj.bias'}, {'critic_backbone.layers.0.self_attn.o_proj.weight', 'value_model.model.layers.0.self_attn.o_proj.weight'}, {'value_model.model.layers.0.mlp.gate_proj.weight', 'critic_backbone.layers.0.mlp.gate_proj.weight'}, {'critic_backbone.layers.0.mlp.up_proj.weight', 'value_model.model.layers.0.mlp.up_proj.weight'}, {'critic_backbone.layers.0.mlp.down_proj.weight', 'value_model.model.layers.0.mlp.down_proj.weight'}, {'value_model.model.layers.0.input_layernorm.weight', 'critic_backbone.layers.0.input_layernorm.weight'}, {'value_model.model.layers.0.post_attention_layernorm.weight', 'critic_backbone.layers.0.post_attention_layernorm.weight'}, {'critic_backbone.layers.1.self_attn.q_proj.weight', 'value_model.model.layers.1.self_attn.q_proj.weight'}, {'critic_backbone.layers.1.self_attn.q_proj.bias', 'value_model.model.layers.1.self_attn.q_proj.bias'}, {'critic_backbone.layers.1.self_attn.k_proj.weight', 'value_model.model.layers.1.self_attn.k_proj.weight'}, {'value_model.model.layers.1.self_attn.k_proj.bias', 'critic_backbone.layers.1.self_attn.k_proj.bias'}, {'critic_backbone.layers.1.self_attn.v_proj.weight', 'value_model.model.layers.1.self_attn.v_proj.weight'}, {'critic_backbone.layers.1.self_attn.v_proj.bias', 'value_model.model.layers.1.self_attn.v_proj.bias'}, {'critic_backbone.layers.1.self_attn.o_proj.weight', 'value_model.model.layers.1.self_attn.o_proj.weight'}, {'critic_backbone.layers.1.mlp.gate_proj.weight', 'value_model.model.layers.1.mlp.gate_proj.weight'}, {'critic_backbone.layers.1.mlp.up_proj.weight', 'value_model.model.layers.1.mlp.up_proj.weight'}, {'critic_backbone.layers.1.mlp.down_proj.weight', 'value_model.model.layers.1.mlp.down_proj.weight'}, {'critic_backbone.layers.1.input_layernorm.weight', 'value_model.model.layers.1.input_layernorm.weight'}, {'critic_backbone.layers.1.post_attention_layernorm.weight', 'value_model.model.layers.1.post_attention_layernorm.weight'}, {'value_model.model.norm.weight', 'critic_backbone.norm.weight'}].
              A potential way to correctly save your model is to use `save_model`.
              More information at https://huggingface.co/docs/safetensors/torch_shared_tensors
  FAILED tests/experimental/test_ppo_trainer.py::TestPPOTrainer::test_basic_training - RuntimeError: 
              Some tensors share memory, this will lead to duplicate memory on disk and potential differences when loading them again: [{'value_model.model.embed_tokens.weight', 'critic_backbone.embed_tokens.weight'}, {'value_model.model.layers.0.self_attn.q_proj.weight', 'critic_backbone.layers.0.self_attn.q_proj.weight'}, {'value_model.model.layers.0.self_attn.q_proj.bias', 'critic_backbone.layers.0.self_attn.q_proj.bias'}, {'critic_backbone.layers.0.self_attn.k_proj.weight', 'value_model.model.layers.0.self_attn.k_proj.weight'}, {'value_model.model.layers.0.self_attn.k_proj.bias', 'critic_backbone.layers.0.self_attn.k_proj.bias'}, {'critic_backbone.layers.0.self_attn.v_proj.weight', 'value_model.model.layers.0.self_attn.v_proj.weight'}, {'critic_backbone.layers.0.self_attn.v_proj.bias', 'value_model.model.layers.0.self_attn.v_proj.bias'}, {'critic_backbone.layers.0.self_attn.o_proj.weight', 'value_model.model.layers.0.self_attn.o_proj.weight'}, {'value_model.model.layers.0.mlp.gate_proj.weight', 'critic_backbone.layers.0.mlp.gate_proj.weight'}, {'value_model.model.layers.0.mlp.up_proj.weight', 'critic_backbone.layers.0.mlp.up_proj.weight'}, {'critic_backbone.layers.0.mlp.down_proj.weight', 'value_model.model.layers.0.mlp.down_proj.weight'}, {'value_model.model.layers.0.input_layernorm.weight', 'critic_backbone.layers.0.input_layernorm.weight'}, {'critic_backbone.layers.0.post_attention_layernorm.weight', 'value_model.model.layers.0.post_attention_layernorm.weight'}, {'value_model.model.layers.1.self_attn.q_proj.weight', 'critic_backbone.layers.1.self_attn.q_proj.weight'}, {'critic_backbone.layers.1.self_attn.q_proj.bias', 'value_model.model.layers.1.self_attn.q_proj.bias'}, {'value_model.model.layers.1.self_attn.k_proj.weight', 'critic_backbone.layers.1.self_attn.k_proj.weight'}, {'critic_backbone.layers.1.self_attn.k_proj.bias', 'value_model.model.layers.1.self_attn.k_proj.bias'}, {'value_model.model.layers.1.self_attn.v_proj.weight', 'critic_backbone.layers.1.self_attn.v_proj.weight'}, {'critic_backbone.layers.1.self_attn.v_proj.bias', 'value_model.model.layers.1.self_attn.v_proj.bias'}, {'value_model.model.layers.1.self_attn.o_proj.weight', 'critic_backbone.layers.1.self_attn.o_proj.weight'}, {'critic_backbone.layers.1.mlp.gate_proj.weight', 'value_model.model.layers.1.mlp.gate_proj.weight'}, {'value_model.model.layers.1.mlp.up_proj.weight', 'critic_backbone.layers.1.mlp.up_proj.weight'}, {'critic_backbone.layers.1.mlp.down_proj.weight', 'value_model.model.layers.1.mlp.down_proj.weight'}, {'critic_backbone.layers.1.input_layernorm.weight', 'value_model.model.layers.1.input_layernorm.weight'}, {'critic_backbone.layers.1.post_attention_layernorm.weight', 'value_model.model.layers.1.post_attention_layernorm.weight'}, {'value_model.model.norm.weight', 'critic_backbone.norm.weight'}].
              A potential way to correctly save your model is to use `save_model`.
              More information at https://huggingface.co/docs/safetensors/torch_shared_tensors
  = 2 failed,
```

Stacktrace:
```python
   >       trainer.train()
  
  tests/experimental/test_ppo_trainer.py:747: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  trl/experimental/ppo/ppo_trainer.py:912: in train
      self._save_checkpoint(model, trial=None)
  trl/experimental/ppo/ppo_trainer.py:1027: in _save_checkpoint
      super()._save_checkpoint(model, trial)
  .venv/lib/python3.13/site-packages/transformers/trainer.py:3038: in _save_checkpoint
      self.save_model(output_dir, _internal_call=True)
  trl/experimental/ppo/ppo_trainer.py:595: in save_model
      super().save_model(output_dir, _internal_call)
  .venv/lib/python3.13/site-packages/transformers/trainer.py:3786: in save_model
      self._save(output_dir)
  .venv/lib/python3.13/site-packages/transformers/trainer.py:3812: in _save
      safetensors.torch.save_file(
  .venv/lib/python3.13/site-packages/safetensors/torch.py:307: in save_file
      serialize_file(_flatten(tensors), filename, metadata=metadata)
                     ^^^^^^^^^^^^^^^^^
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  tensors = OrderedDict({'policy.model.embed_tokens.weight': tensor([[ 0.0234,  0.0266, -0.0123,  ..., -0.0057,  0.0125, -0.0255],...ensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
         device='cuda:0', dtype=torch.bfloat16)})
  
      def _flatten(tensors: Dict[str, torch.Tensor]) -> Dict[str, Dict[str, Any]]:
          if not isinstance(tensors, dict):
              raise ValueError(
                  f"Expected a dict of [str, torch.Tensor] but received {type(tensors)}"
              )
      
          invalid_tensors = []
          for k, v in tensors.items():
              if not isinstance(v, torch.Tensor):
                  raise ValueError(
                      f"Key `{k}` is invalid, expected torch.Tensor but received {type(v)}"
                  )
      
              if v.layout != torch.strided:
                  invalid_tensors.append(k)
          if invalid_tensors:
              raise ValueError(
                  f"You are trying to save a sparse tensors: `{invalid_tensors}` which this library does not support."
                  " You can make it a dense tensor before saving with `.to_dense()` but be aware this might"
                  " make a much larger file than needed."
              )
      
          shared_pointers = _find_shared_tensors(tensors)
          failing = []
          for names in shared_pointers:
              if len(names) > 1:
                  failing.append(names)
      
          if failing:
  >           raise RuntimeError(
                  f"""
                  Some tensors share memory, this will lead to duplicate memory on disk and potential differences when loading them again: {failing}.
                  A potential way to correctly save your model is to use `save_model`.
                  More information at https://huggingface.co/docs/safetensors/torch_shared_tensors
                  """
              )
  E           RuntimeError: 
  E                       Some tensors share memory, this will lead to duplicate memory on disk and potential differences when loading them again: [{'value_model.model.embed_tokens.weight', 'critic_backbone.embed_tokens.weight'}, {'value_model.model.layers.0.self_attn.q_proj.weight', 'critic_backbone.layers.0.self_attn.q_proj.weight'}, {'value_model.model.layers.0.self_attn.q_proj.bias', 'critic_backbone.layers.0.self_attn.q_proj.bias'}, {'critic_backbone.layers.0.self_attn.k_proj.weight', 'value_model.model.layers.0.self_attn.k_proj.weight'}, {'value_model.model.layers.0.self_attn.k_proj.bias', 'critic_backbone.layers.0.self_attn.k_proj.bias'}, {'critic_backbone.layers.0.self_attn.v_proj.weight', 'value_model.model.layers.0.self_attn.v_proj.weight'}, {'critic_backbone.layers.0.self_attn.v_proj.bias', 'value_model.model.layers.0.self_attn.v_proj.bias'}, {'critic_backbone.layers.0.self_attn.o_proj.weight', 'value_model.model.layers.0.self_attn.o_proj.weight'}, {'value_model.model.layers.0.mlp.gate_proj.weight', 'critic_backbone.layers.0.mlp.gate_proj.weight'}, {'value_model.model.layers.0.mlp.up_proj.weight', 'critic_backbone.layers.0.mlp.up_proj.weight'}, {'critic_backbone.layers.0.mlp.down_proj.weight', 'value_model.model.layers.0.mlp.down_proj.weight'}, {'value_model.model.layers.0.input_layernorm.weight', 'critic_backbone.layers.0.input_layernorm.weight'}, {'critic_backbone.layers.0.post_attention_layernorm.weight', 'value_model.model.layers.0.post_attention_layernorm.weight'}, {'value_model.model.layers.1.self_attn.q_proj.weight', 'critic_backbone.layers.1.self_attn.q_proj.weight'}, {'critic_backbone.layers.1.self_attn.q_proj.bias', 'value_model.model.layers.1.self_attn.q_proj.bias'}, {'value_model.model.layers.1.self_attn.k_proj.weight', 'critic_backbone.layers.1.self_attn.k_proj.weight'}, {'critic_backbone.layers.1.self_attn.k_proj.bias', 'value_model.model.layers.1.self_attn.k_proj.bias'}, {'value_model.model.layers.1.self_attn.v_proj.weight', 'critic_backbone.layers.1.self_attn.v_proj.weight'}, {'critic_backbone.layers.1.self_attn.v_proj.bias', 'value_model.model.layers.1.self_attn.v_proj.bias'}, {'value_model.model.layers.1.self_attn.o_proj.weight', 'critic_backbone.layers.1.self_attn.o_proj.weight'}, {'critic_backbone.layers.1.mlp.gate_proj.weight', 'value_model.model.layers.1.mlp.gate_proj.weight'}, {'value_model.model.layers.1.mlp.up_proj.weight', 'critic_backbone.layers.1.mlp.up_proj.weight'}, {'critic_backbone.layers.1.mlp.down_proj.weight', 'value_model.model.layers.1.mlp.down_proj.weight'}, {'critic_backbone.layers.1.input_layernorm.weight', 'value_model.model.layers.1.input_layernorm.weight'}, {'critic_backbone.layers.1.post_attention_layernorm.weight', 'value_model.model.layers.1.post_attention_layernorm.weight'}, {'value_model.model.norm.weight', 'critic_backbone.norm.weight'}].
  E                       A potential way to correctly save your model is to use `save_model`.
  E                       More information at https://huggingface.co/docs/safetensors/torch_shared_tensors
  
  .venv/lib/python3.13/site-packages/safetensors/torch.py:535: RuntimeError
```

--- a/trl/experimental/ppo/ppo_trainer.py
+++ b/trl/experimental/ppo/ppo_trainer.py
@@ -584,20 +584,18 @@ class PPOTrainer(BaseTrainer):
                 self.model.policy.set_adapter(self.model_adapter_name or "default")
 
     def save_model(self, output_dir: str | None = None, _internal_call: bool = False):
-        if not _internal_call:
-            backup_model = self.model
-            if hasattr(self.model, "policy"):
-                self.model = self.model.policy  # save only the policy for inference
-            if self.is_deepspeed_enabled:
-                backup_deepspeed = self.deepspeed
-                self.deepspeed = self.model
+        backup_model = self.model
+        if hasattr(self.model, "policy"):
+            self.model = self.model.policy  # save only the policy for inference
+        if self.is_deepspeed_enabled:
+            backup_deepspeed = self.deepspeed
+            self.deepspeed = self.model
 
         super().save_model(output_dir, _internal_call)
 
-        if not _internal_call:
-            self.model = backup_model
-            if self.is_deepspeed_enabled:
-                self.deepspeed = backup_deepspeed
+        self.model = backup_model
+        if self.is_deepspeed_enabled:
+            self.deepspeed = backup_deepspeed
 
     def train(self):
         args = self.args

